foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/oxm/XMLMarshaller.java

```diff
@@ -1396,4 +1401,19 @@ public Object getMarshalAttributeGroup() {
         return this.marshalAttributeGroup;
     }
 
+    public Boolean isLogPayload() {
+        return logPayload;
+    }
+
+    public void setLogPayload(Boolean logPayload) {
+        this.logPayload = logPayload;
+    }
+
+    private void logMarshall(Object object) {
+        String i18nmsg = JAXBLocalization.buildMessage("start_marshalling", new Object[] { (object!= null)?object.getClass().getName():"N/A", this.mediaType });
```

Comment:
log method is (or should be) capable of accepting "start_marshalling" string and taking it from the resource bundle if needed (see the overridden impl taking boolean translate argument), so you don't have to construct the message if log level is ie INFO (this has performance impact)